tests/modules/moduleconfigs.c

```diff
@@ -121,13 +121,13 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
     }
 
     /* On the stack to make sure we're copying them. */
-    const char *enum_vals[] = {"none", "one", "two", "three"};
-    const int int_vals[] = {0, 1, 2, 4};
+    const char *enum_vals[] = {"none", "five", "one", "two", "four"};
+    const int int_vals[] = {0, 5, 1, 2, 4};
 
-    if (RedisModule_RegisterEnumConfig(ctx, "enum", 1, REDISMODULE_CONFIG_DEFAULT, enum_vals, int_vals, 4, getEnumConfigCommand, setEnumConfigCommand, NULL, NULL) == REDISMODULE_ERR) {
+    if (RedisModule_RegisterEnumConfig(ctx, "enum", 1, REDISMODULE_CONFIG_DEFAULT, enum_vals, int_vals, 5, getEnumConfigCommand, setEnumConfigCommand, NULL, NULL) == REDISMODULE_ERR) {
         return REDISMODULE_ERR;
     }
-    if (RedisModule_RegisterEnumConfig(ctx, "flags", 3, REDISMODULE_CONFIG_DEFAULT | REDISMODULE_CONFIG_BITFLAGS, enum_vals, int_vals, 4, getFlagsConfigCommand, setFlagsConfigCommand, NULL, NULL) == REDISMODULE_ERR) {
+    if (RedisModule_RegisterEnumConfig(ctx, "flags", 3, REDISMODULE_CONFIG_DEFAULT | REDISMODULE_CONFIG_BITFLAGS, enum_vals, int_vals, 5, getFlagsConfigCommand, setFlagsConfigCommand, NULL, NULL) == REDISMODULE_ERR) {
```

Comment:
This seems odd, why is it possible to use `REDISMODULE_CONFIG_BITFLAGS` and still allow enum overlap, should we validate it and return `REDISMODULE_ERR`?

Reply:
by overlap you mean like 4 and 5 overlap? or an identical match like an alias (`four` and `quattro`)
i don't see the problem, please be more specific.

Reply:
Like 4 and 5.
I think allowing 5 would cause more confusion, is it possible that the module developers defined the following enums and used `REDISMODULE_CONFIG_BITFLAGS`, maybe we should be able to avoid this error.
At least we allow `config set enum 'one two'`, why should we support `config set enum three`?
```
apple: 1
banana: 2
orange: 3 -> Actually it is 4
```

Reply:
@sundb the whole purpose of this PR is to fix problem with overlapping enum bits that i was overlooking when i created the `REDISMODULE_CONFIG_BITFLAGS` few weeks ago (part of 7.0.0 GA).
so are you arguing that instead of all the changes i do here, i should just forbid overlapping enum bits?

Note that the whole RM_RegisterEnumConfig feature is new in 7.0, and so is the BITFLAGS feature.
people with values such as [apple=1,banana=2,orange=3] should be using plain RM_RegisterEnumConfig, and ones that wanna use BITFLAGS should use [1,2,4], in which case they get two useful features:
1. being able to define multiple bits separately (e.g. `config set myconfig "one four"`)
2. still able to use an alias for a common set of bits (e.g. `config set myconfig "five"`)

there's no breaking change here since this is a completely new feature with 0 adoption.
since the default for RM_RegisterEnumConfig is without bits, i think the risk of someone using [apple=1,banana=2,orange=3] with it is low (most people will create normal enums and be unaware of this feature).

or maybe i'm still missing your point?

Reply:
> so are you arguing that instead of all the changes i do here, i should just forbid overlapping enum bits?

    Yes, until this PR I thought that enum should be [0, 1, 2, 4, ...] when using `REDISMODULE_CONFIG_BITFLAGS`, I personally feel that `config set myconfig "five"` dose not convenient, but a bit redundant.

Reply:
Well, Yossi gave the same comment in https://github.com/redis/redis/pull/10594#issuecomment-1108940639 before i explained that this was intended (and didn't bother to test it really works).

Note that the `maxmemory-policy` enum is also in theory somewhat a bit flags thing (or at least could have been).
i.e. it could have been using `allkeys random` instead of `allkeys-random`, but even if we did that, i think some special aliases, possibly matching multiple bits could be nice.